samples/extensions/extended_dynamic_state2/extended_dynamic_state2.cpp

```diff
@@ -0,0 +1,1065 @@
+	auto &requested_extended_dynamic_state2_features                                   = gpu.request_extension_features<VkPhysicalDeviceExtendedDynamicState2FeaturesEXT>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTENDED_DYNAMIC_STATE_2_FEATURES_EXT);
+	requested_extended_dynamic_state2_features.extendedDynamicState2                   = VK_TRUE;
+	requested_extended_dynamic_state2_features.extendedDynamicState2PatchControlPoints = VK_TRUE;
+
+	auto &requested_extended_dynamic_state_feature                = gpu.request_extension_features<VkPhysicalDeviceExtendedDynamicStateFeaturesEXT>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTENDED_DYNAMIC_STATE_FEATURES_EXT);
+	requested_extended_dynamic_state_feature.extendedDynamicState = VK_TRUE;
+
+	// Tessellation shader support is required for this example
+	auto &requested_features = gpu.get_mutable_requested_features();
+	if (gpu.get_features().tessellationShader)
+	{
+		requested_features.tessellationShader = VK_TRUE;
+	}
+	else
+	{
+		throw vkb::VulkanException(VK_ERROR_FEATURE_NOT_PRESENT, "Selected GPU does not support tessellation shaders!");
+	}
+
+	if (gpu.get_features().fillModeNonSolid)
+	{
+		requested_features.fillModeNonSolid = VK_TRUE;
+	}
+
+	if (gpu.get_features().samplerAnisotropy)
+	{
+		gpu.get_mutable_requested_features().samplerAnisotropy = true;
+	}
+}
+
+/**
+ * @fn void ExtendedDynamicState2::on_update_ui_overlay(vkb::Drawer &drawer)
+ * @brief Projecting GUI and transferring data between GUI and app
+ */
+void ExtendedDynamicState2::on_update_ui_overlay(vkb::Drawer &drawer)
+{
+	if (drawer.header("Settings"))
+	{
+		if (drawer.checkbox("Tessellation Enable", &gui_settings.tessellation))
+		{
+			update_uniform_buffers();
+		}
+		if (drawer.input_float("Tessellation Factor", &gui_settings.tess_factor, 1.0f, 1))
+		{
+			update_uniform_buffers();
+		}
+
+		if (drawer.input_float("Patch Control Points", &gui_settings.patch_control_points_float, 1.0f, 1))
+		{
+			if (gui_settings.patch_control_points_float < 1)
+			{
+				gui_settings.patch_control_points_float = 1;
+			}
+			gui_settings.patch_control_points = static_cast<int>(roundf(gui_settings.patch_control_points_float));
+		}
+	}
+	if (drawer.header("Models"))
+	{
+		const char *col_names = {"Name"};
+		if (drawer.checkbox("Selection effect active", &gui_settings.selection_active))
+		{
+		}
+		ImGui::Columns(2, "Name");
+		ImGui::SetColumnWidth(0, 150);
+		int                       obj_cnt = scene_nodes.at(SCENE_BASELINE_OBJ_INDEX).size();
+		std::vector<const char *> obj_names;
+
+		for (int i = 0; i < obj_cnt; i++)
+		{
+			obj_names.push_back((scene_nodes.at(SCENE_BASELINE_OBJ_INDEX).at(i).name).c_str());
+		}
+		ImGui::ListBox("", &gui_settings.selected_obj, obj_names.data(), obj_cnt);
+		ImGui::NextColumn();
+		if (drawer.checkbox("Depth Bias Enable", &gui_settings.objects[gui_settings.selected_obj].depth_bias))
+		{
+		}
+		if (drawer.checkbox("Rasterizer Discard", &gui_settings.objects[gui_settings.selected_obj].rasterizer_discard))
```

Comment:
Same here, the if statement is unnecessary.

Reply:
fixed